src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -325,8 +325,8 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
                                 case ConversionKind.ImplicitUserDefined:
                                     return false;
                                 default:
-                                    Debug.Assert(false, "Unhandled conversion kind in reordering logic");
-                                    return false;
+                                    // Unhandled conversion kind in reordering logic
```

Comment:
I think returning false might be acceptable in this case. It is ok to say that something is not safe to reorder.